### DISCUSSION TITLE
at函数原型

### DISCUSSION THREAD

**Discussion by Mr-JingShi** [hexhacking/xUnwind#7] [Q&A] [answered]
请问下：std::string里at的函数原型是reference at( size_type pos );
xUnwind里怎么就变成了typedef const char * (*xu_libcxx_string_at_t)(void *self, size_t pos);
而且运行起来没有问题，百思不得其解。

请老师解答下。

**Comment by Mr-JingShi** [comment]
> 请问下：std::string里at的函数原型是reference at( size_type pos ); xUnwind里怎么就变成了typedef const char * (*xu_libcxx_string_at_t)(void *self, size_t pos); 而且运行起来没有问题，百思不得其解。
> 
> 请老师解答下。

c++里的reference底层实现是相对应类型的指针的原因吗？

**Comment by caikelun** [comment, marked as answer]
在运行时，根据C++成员函数的调用约定，需要把this作为第一个参数传递。如果是C++代码，C++编译器会为我们完成这些（可以看下C++函数调用生成的汇编代码）。但是在C代码中我们需要手工传递this参数。
